woocommerce/woocommerce-ios
# [Order form] Extract ProductStepper from ProductRow, and use it only in views where it is needed

Closes: #11368 
Part of #11357 


## Why


Now that the `ProductStepper` has its own view model after the previous PR https://github.com/woocommerce/woocommerce-ios/pull/11362, the `ProductStepper` view can now be extracted from  the parent view `ProductRow` so that the UI/UX of each SwiftUI view is clearly separated without a lot of conditional logic. Sorry for the 500+ diffs, a lot of them are simple renaming, and I tried documenting the major changes in the section below.

## How

The major changes were to move the following properties from `ProductRowViewModel` to a new view model `ProductWithQuantityStepperViewModel` for a new view `ProductWithQuantityStepperView` that contains a `ProductRow` and `ProductStepper` like how `ProductRow` used to be when `canChangeQuantity` is `true`.

- `canChangeQuantity: Bool`
- `isReadOnly: Bool`: the bundle product logic was moved to `EditableOrderViewModel`
- `childProductRows: [ProductRowViewModel]`

`ProductWithQuantityStepperView` is used in two cases, `ConfigurableBundleItemView` and `CollapsibleProductCard`. These two views have a few more properties specific to these views like `isConfigurable` that are still in `ProductRowViewModel` or `ProductWithQuantityStepperView`, these will be extracted to their own view models in future subtasks of #11357.

After the changes, `ProductRow` is just a simple view that contains a thumbnail and a V stack of product-related info that is shown in product/variation selector and a few other screens.

Unit tests were updated to fix the interface changes plus the move of properties. I didn't add new tests for `ProductWithQuantityStepperView` since the PR diffs are already large.

## Testing instructions


No changes on the order form UX are expected. The following steps might not cover all the possible cases, please feel free to diverge a bit and share any issues!

- Go to the Orders tab
- Tap on `+` to create an order
- Tap `Add Products` --> the product rows in the product selector should behave the same as before
- Search/select at least one product and tap the bottom CTA to confirm the selection --> the selected product(s) should show quantity as 1
- Expand any product card
- Tap to increment the product quantity --> the quantity label in the price in the product card should also be updated
- After the payments section is synced, tap `Create` to create the order --> the created order should have the recently set quantity for each product
- Tap `Edit` to edit the order --> the quantity for each product should match what was set previously
- Tap to increment the product quantity --> after syncing, the quantity label in the price in the product card should be updated
- Tap to decrease the product quantity to 0 --> the product should be removed
- Tap `Done` --> the updated order should have the recently set quantity for each product 

---

- [ ] @jaclync tests the product bundle configuration flow for order creation
- [ ] @jaclync tests the product bundle configuration flow for order editing

---
